src/MorseL.Client/Connection.cs

```diff
@@ -491,7 +492,18 @@ public async Task DisposeAsync(CancellationToken ct = default(CancellationToken)
                 || _clientWebSocket.State == WebSocketState.CloseReceived
                 || _clientWebSocket.State == WebSocketState.CloseSent)
             {
-                await _clientWebSocket.CloseAsync(WebSocketCloseStatus.Empty, null, ct).ConfigureAwait(false);
+                try
+                {
+                    await _clientWebSocket.CloseAsync(WebSocketCloseStatus.Empty, null, ct).ConfigureAwait(false);
+                }
+                catch (IOException ex) when (ex.InnerException is SocketException)
+                {
+                    // Ignore SocketExceptions intentionally when closing a socket. The main reason for this
+                    // is issues encountered when losing network connection. Closing a socket after losing
+                    // network connection results in a "network subsystem is down" exception. It tries to
+                    // send a close message but cannot because there is no connection anymore.
+                    _logger.LogWarning(new EventId(), ex, "Observed SocketException - likely OK");
```

Comment:
I don't think we need an event ID here, just use the overload without it.

Reply:
Also, we may want to save this Exception and pass it below in the FireClosed() call. I can't add a comment down there...not sure why, but it's on line 517.

Reply:
> 
> 
> I don't think we need an event ID here, just use the overload without it.

Not sure what version that overload was added, but MorseL is using v1.1.2 of Microsoft.Extensions.Logging.Abstractions which does not have that overload.

Reply:
> 
> 
> Also, we may want to save this Exception and pass it below in the FireClosed() call. I can't add a comment down there...not sure why, but it's on line 517.

I can add that. Didn't notice it took an exception as a parameter.

Reply:
Hm, perhaps you need it if you log out an exception.